lib/python/Pyxsim/__init__.py

```diff
@@ -63,58 +62,40 @@ def _build(
         for key in env:
             my_env[key] = str(env[key])
 
-    # The build process differs between cmake and xmake; cater for both
-    if cmake:
-        make_cmd = ["cmake", "-B", f"bin/{binary_child}"]
-        build_cmd = ["cmake", "--build", f"bin/{binary_child}"]
-
-        if clean_only:
-            build_cmd += ["--target", "clean"]
-            do_clean = False
-        if do_clean:
-            build_cmd += ["--clean-first"]
+    cmd = ["xmake"]
+    if clean_only:
+        cmd += ["clean"]
+        do_clean = False
 
-        if silent:
-            output_args = {"stderr":subprocess.DEVNULL, "stdout":subprocess.DEVNULL}
-        else:
-            output_args = {"stderr":subprocess.STDOUT}
+    if do_clean:
+        call_get_output(["xmake", "clean"], cwd=path, env=my_env)
 
-        subprocess.run(make_cmd, cwd=path, env=my_env, **output_args)
-        subprocess.run(build_cmd, cwd=path, env=my_env, **output_args)
-
-        return (True, "Subprocesses run")
+    if cmake:
+        if not clean_only:
+            cmd += [Path(xe_path).stem]
     else:
-        if clean_only:
-            cmd = ["xmake", "clean"]
-            do_clean = False
-        else:
-            cmd = ["xmake", "all"]
-
-        if do_clean:
-            call_get_output(["xmake", "clean"], cwd=path, env=my_env)
-
         if build_config is not None:
```

Comment:
isn't the cmake execution path missing something like cmd +=cmd += ["%s" % build_config]?